src/TextBoxSearch/components/TextBoxSearchContainer.ts

```diff
@@ -120,36 +183,46 @@ export default class SearchContainer extends Component<ContainerProps, Container
                 operator: "or"
             };
         }
-
         if (targetListView && targetListView._datasource && searchQuery) {
             const constraints: string[] = [];
-            const dayInMilliSeconds = 24 * 60 * 60 * 1000;
+
             this.props.attributeList.forEach(searchAttribute => {
-                const { attribute } = searchAttribute;
-                if (mx.meta.getEntity(this.props.entity).isDate(attribute)) {
-                    let finalDate = window.mx.parser.parseValue(searchQuery, "datetime", { selector: "date" });
-                    const isLocalized = mx.meta.getEntity(this.props.entity).isLocalizedDate(attribute);
-                    if (finalDate) {
-                        if (!isLocalized) {
-                            const delocalizedTime = window.mx.parser.delocalizeEpoch(finalDate);
-                            finalDate = new Date(delocalizedTime);
-                        }
-                        const currentDate = finalDate.getTime();
-                        const nextDate = finalDate.getTime() + dayInMilliSeconds;
-                        const constraint = `(${attribute}>=${currentDate} and ${attribute}<${nextDate})`;
-                        constraints.push(constraint);
-                    }
-                } else {
-                    constraints.push(`contains(${attribute},'${searchQuery}')`);
-                }
+                constraints.push(`contains(${searchAttribute.attribute},'${searchQuery}')`);
             });
 
-            return constraints.length ? "[" + constraints.join(" or ") + "]" : "";
+            return "[" + constraints.join(" or ") + "]";
         }
         return "";
     }
 
-    private connectToListView() {
+    private getDateConstraints = () => {
+        const { datePicker, searchNode } = this.state;
+        const constraints: string[] = [];
+        const dayInMilliSeconds = 24 * 60 * 60 * 1000;
+
+        if (datePicker) {
+            if (datePicker.get("value")) {
+                this.props.attributeList.forEach(searchAttribute => {
+                    const { attribute } = searchAttribute;
+                    let finalDate = (datePicker.get("value") as Date).getTime();
+                    if (!mx.meta.getEntity(this.props.entity).isLocalizedDate(attribute)) {
+                        finalDate = window.mx.parser.delocalizeEpoch(datePicker.get("value") as Date);
+                    }
+                    const nextDate = finalDate + dayInMilliSeconds;
+                    constraints.push(`(${attribute}>=${finalDate} and ${attribute}<${nextDate})`);
+                });
+                if (this.state.alertMessage) {
+                    this.setState({ alertMessage: "" });
+                }
+            } else if (searchNode.value) {
```

Comment:
`searchNode` is `HTMLInputElement | undefined` i recommend to change to `else if (searchNode && searchNode.value) {`